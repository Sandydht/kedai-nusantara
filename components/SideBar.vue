<template>
  <div id="sideBar" @click="click">
    <nav-bar />
  </div>
</template>

<script>
import NavBar from '~/components/NavBar.vue'

export default {
  name: 'SideBar',
  components: {
    NavBar
  },
  emits: ['handleClick'],
  methods: {
    click () {
      this.$emit('handleClick')
    }
  }
}
</script>
