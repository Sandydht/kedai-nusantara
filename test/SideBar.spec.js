import { shallowMount } from '@vue/test-utils'
import SideBar from '~/components/SideBar.vue'

describe('SideBar.vue', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(SideBar)
  })

  test('should emits an event when clicked', async () => {
    await wrapper.find('div').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('handleClick')
  })

  test('has expected html structure', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
