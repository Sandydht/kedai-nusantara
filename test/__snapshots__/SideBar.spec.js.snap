// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar.vue has expected html structure 1`] = `
"<div id=\\"sideBar\\">
  <nav-bar-stub></nav-bar-stub>
</div>"
`;
